packages/search-in-workspace/src/browser/search-in-workspace-widget.tsx

```diff
@@ -316,19 +316,30 @@ export class SearchInWorkspaceWidget extends BaseWidget implements StatefulWidge
 
     protected renderReplaceFieldToggle(): React.ReactNode {
         const toggle = <span className={`fa fa-caret-${this.showReplaceField ? 'down' : 'right'}`}></span>;
+
+        const replaceToggle = () => {
+            const elArr = document.getElementsByClassName('replace-toggle');
+            if (elArr && elArr.length > 0) {
+                (elArr[0] as HTMLElement).focus();
+            }
+            this.showReplaceField = !this.showReplaceField;
+            this.resultTreeWidget.showReplaceButtons = this.showReplaceField;
+            this.update();
+        };
+
         return <div
             title='Toggle Replace'
             className='replace-toggle'
             tabIndex={0}
-            onClick={e => {
-                const elArr = document.getElementsByClassName('replace-toggle');
-                if (elArr && elArr.length > 0) {
-                    (elArr[0] as HTMLElement).focus();
+            onClick={() => {
+                replaceToggle();
```

Comment:
https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#no-bind-fn-in-event-handlers please review other places